substrate/frame/contracts/src/exec.rs

```diff
@@ -1217,16 +1225,25 @@ where
 		value: BalanceOf<T>,
 		input_data: Vec<u8>,
 		allows_reentry: bool,
+		read_only: bool,
 	) -> Result<ExecReturnValue, ExecError> {
 		// Before pushing the new frame: Protect the caller contract against reentrancy attacks.
 		// It is important to do this before calling `allows_reentry` so that a direct recursion
 		// is caught by it.
 		self.top_frame_mut().allows_reentry = allows_reentry;
 
+		// Enable read-only access if requested; cannot disable it if already set.
+		let frame_read_only = read_only || self.top_frame().read_only;
```

Comment:
```suggestion
		let read_only = read_only || self.top_frame().read_only;
```

We can just shadow here to avoid the mental load of another binding.